pkg/apis/softwarecomposition/types.go

```diff
@@ -216,3 +216,109 @@ type VulnerabilityManifestSummaryList struct {
 
 	Items []VulnerabilityManifestSummary
 }
+
+// -------------- configuration scanning types -----------------
+
+// WorkloadConfigurationScan is a list of workload configuration scan results.
+type WorkloadConfigurationScanList struct {
+	metav1.TypeMeta
+	metav1.ListMeta
+
+	Items []WorkloadConfigurationScan
+}
+
+// WorkloadConfigurationScan is a custom resource that describes a configuration scan result of a workload.
+type WorkloadConfigurationScan struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec WorkloadConfigurationScanSpec
+}
+
+type WorkloadConfigurationScanSpec struct {
+	Controls       map[string]ScannedControl
+	RelatedObjects []WorkloadScanRelatedObject
+}
+
+type ScannedControl struct {
+	ControlID string
+	Name      string
+	Severity  ControlSeverity
+	Status    ScannedControlStatus
+	Rules     []ScannedControlRule
```

Comment:
question: storing rules inside objects is the only way we can proceed now, right?

The alternative would be storing these definitions somewhere inside the cluster, and we have decided we are not doing this at the moment. Is that correct?

Reply:
Resolved on our meeting: yes, we absolutely need this at the moment, and this field stays.